WordPressAuthenticator/Credentials/WordPressCredentials.swift

```diff
@@ -9,5 +9,5 @@ public enum WordPressCredentials {
 
     /// WordPress.com Site Credentials.
     ///
-    case wpcom(authToken: String, isJetpackLogin: Bool, multifactor: Bool)
+    case wpcom(authToken: String, isJetpackLogin: Bool, multifactor: Bool, siteURL: String)
```

Comment:
It pains me that we can assign default values to enums.  

It might be nice to have the default `dotcomAddress` declared here vs in `LoginFieldsMeta` since its a credentials thing, and it might be nice if we could mimic the behavior of default values with an internal builder class.   It would look something like (just psuedo code)
```
public enum WordPressCredentials {
    public struct Builder {
         public wpcom(authToken: String, isJetpackLogin: Bool, multifactor: Bool, siteURL: String?) {
                let site = siteURL ?? "http://wordpress.com"
                return WordPressCredentials.wpcom(authToken, isJetpackLogin, multifactor, site)
          }
    }
   [...]
}
```
And then wherever wpcom credentials were used: 
```
WordPressCredentials.Builder.wpcom(...) 
```

Just a thought.  This or something like it could save all the different places where the site is being checked before deciding whether to use the default or not.  Up to you!